python/cudf/cudf/core/frame.py

```diff
@@ -97,6 +97,133 @@ def _as_column(self):
+
+       """
+        num_rows = len(self)
+        if num_rows == 0:
+            return self
+        start, stop, stride = arg.indices(num_rows)
+        if start < 0:
+            start = start + num_rows
+        if stop < 0:
+            stop = stop + num_rows
+
+        if start > stop:
+            return self._empty_like()
+        else:
+            start = len(self) if start > num_rows else start
+            stop = len(self) if stop > num_rows else stop
+
+            result = self._from_table(
+                libcudfxx.copying.table_slice(self, [start, stop])[0]
+            )
+
+            result._copy_categories(self)
+            return result
+
+    def _normalize_columns_and_scalars_type(self, other):
+        if isinstance(other, cudf.DataFrame):
+            return [
+                other[other_col].astype(self[self_col].dtype)._column
+                for self_col, other_col in zip(self.columns, other.columns)
+            ]
+
+        elif isinstance(other, (cudf.Series, cudf.Index)):
+            return other.astype(self.dtype)._column
+
+        else:
+            if is_scalar(other):
+                if (other is not None and not np.isnan(other)) and (
+                    self.dtype.type(other) != other
+                ):
+                    raise TypeError(
+                        "Cannot safely cast non-equivalent {} to {}".format(
+                            type(other).__name__, self.dtype.name
+                        )
+                    )
+
+                return (
+                    self.dtype.type(other)
+                    if (other is not None and not np.isnan(other))
+                    else other
+                )
+            else:
+                out = []
+                for in_col_name, sclr in zip(self.columns, other):
+                    common_dtype = self[in_col_name].dtype
+                    if (
+                        sclr is not None
+                        and not np.isnan(sclr)
+                        and (common_dtype.type(sclr) != sclr)
+                    ):
+                        raise TypeError(
+                            """Cannot safely cast non-equivalent {} to
+                            {}""".format(
+                                type(sclr).__name__, common_dtype.dtype.name
+                            )
+                        )
+
+                    out.append(
+                        common_dtype.type(sclr)
+                        if (sclr is not None and not np.isnan(sclr))
+                        else sclr
+                    )
+
+                return out
+
+    def _copy_if_else(self, boolean_mask, other):
+        """
+       Create a new type of column from self and other according to
+       boolean_mask. if boolean_mask is true at row i, then value at
+       row i in self is used else value from other ay row i.
+       """
+
+        if isinstance(self, cudf.DataFrame):
+            raise TypeError("DataFrame is not supported")
```

Comment:
You'd need to guard against MultiIndex here as well.

@shwina any ideas on how to better handle situations where functions only work against single data column frames like Series / Index?